### PR TITLE
[prop types] fix an assignment to prop

### PR DESCRIPTION
**Pull request by DianaSuvorova**
fixes issue #1288

### CODE REVIEW

File: lib/rules/prop-types.js [hunk 1]
```diff
@@ -126,7 +140,7 @@ module.exports = {
         (utils.getParentES6Component() || utils.getParentES5Component()) &&
         node.object.type === 'ThisExpression' && node.property.name === 'props'
       );
-      const isStatelessFunctionUsage = node.object.name === 'props';
+      const isStatelessFunctionUsage = (node.object.name === 'props' && !isAssignmentToProp(node));
```
Comment (ljharb) [inline lib/rules/prop-types.js]: these wrapping parens aren't necessary

**Comment by DianaSuvorova** [comment]
@ljharb updated

**Comment by alexzherdev** [comment]
@ljharb I stumbled upon this test case and traced to this PR. Why should it not fail? Is assignment to a prop not considered usage? I'd expect this to fail with missing prop type for `bar` prop.
```js
function Foo(props) {
  props.bar = 'bar';
  return <div {...props} />;
}
```

**Comment by alexzherdev** [comment]
Ah I see your point. Although the presence of an assignment doesn't necessarily mean that this prop wasn't passed as well. But I guess it's not enough to mark this as usage.

**Comment by ljharb** [comment]
Right, I don't consider it usage, or an untyped prop.

It's weird, but it should still work :-)

**Pull request merged**